script/Deploy.s.sol

```diff
@@ -0,0 +1,199 @@
+import "../contracts/JBFundingCycleStore.sol";
+import "../contracts/JBDirectory.sol";
+import "../contracts/JBTokenStore.sol";
+import "../contracts/JBSplitsStore.sol";
+import "../contracts/JBFundAccessConstraintsStore.sol";
+import "../contracts/JBController3_1.sol";
+import "../contracts/JBTerminalStore.sol";
+import "../contracts/JBMultiTerminal.sol";
+
+contract Deploy is Script {
+    IPermit2 internal constant _PERMIT2 = IPermit2(0x000000000022D473030F116dDEE9F6B43aC78BA3);
+    JBOperatorStore _operatorStore;
+    JBProjects _projects;
+    JBPrices _prices;
+    JBDirectory _directory;
+    JBFundingCycleStore _fundingCycleStore;
+    JBTokenStore _tokenStore;
+    JBSplitsStore _splitsStore;
+    JBFundAccessConstraintsStore _fundAccessConstraintsStore;
+    JBController3_1 _controller;
+    JBTerminalStore _terminalStore;
+    JBMultiTerminal _multiTerminal;
+
+    function _run(address _manager) internal {
+        vm.broadcast();
+        _deployContracts(_manager);
+    }
+
+    function _deployContracts(address _manager) internal {
+        // 1
+        _operatorStore = new JBOperatorStore();
+        // 2
+        _projects = new JBProjects(_operatorStore, _manager);
+        // 3
+        _prices = new JBPrices(_operatorStore, _projects, _manager);
+        address _directoryAddress = addressFrom(address(this), 5);
+        //4
+        _fundingCycleStore = new JBFundingCycleStore(
+            IJBDirectory(_directoryAddress)
+        );
+        // 5
+        _directory = new JBDirectory(
+            _operatorStore,
+            _projects,
+            _fundingCycleStore,
+            address(this)
+        );
+        _tokenStore = new JBTokenStore(
+            _operatorStore,
+            _projects,
+            _directory,
+            _fundingCycleStore
+        );
+        _splitsStore = new JBSplitsStore(_operatorStore, _projects, _directory);
+        _fundAccessConstraintsStore = new JBFundAccessConstraintsStore(
+            _directory
+        );
+        _controller = new JBController3_1(
+            _operatorStore,
+            _projects,
+            _directory,
+            _fundingCycleStore,
+            _tokenStore,
+            _splitsStore,
+            _fundAccessConstraintsStore
+        );
+        _directory.setIsAllowedToSetFirstController(address(_controller), true);
+        _directory.transferOwnership(_manager);
+        _terminalStore = new JBTerminalStore(
+            _directory,
+            _fundingCycleStore,
+            _prices
+        );
+        _multiTerminal = new JBMultiTerminal(
+            _operatorStore,
+            _projects,
+            _directory,
+            _splitsStore,
+            _terminalStore,
+            _PERMIT2,
+            _manager
+        );
+    }
+
+    //https://ethereum.stackexchange.com/questions/24248/how-to-calculate-an-ethereum-contracts-address-during-its-creation-using-the-so
+    function addressFrom(address _origin, uint256 _nonce)
```

Comment:
We should use forge-std.computeCreateAddress(address deployer, uint256 nonce) (this was from before it was included)

Reply:
great!